contracts/src/v0.8/workflow/dev/WorkflowRegistryManager.sol

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {ITypeAndVersion} from "../../shared/interfaces/ITypeAndVersion.sol";
+
+import {Ownable2StepMsgSender} from "../../shared/access/Ownable2StepMsgSender.sol";
+
+/// @title WorkflowRegistryManager
+/// @notice This contract manages the versions of WorkflowRegistry contracts deployed over time.
+/// @dev This contract allows the owner to add, activate, and manage versions of WorkflowRegistry contracts. It tracks
+/// deployment information for each version, including deployment timestamp, chain ID, and active status. Only one
+/// version can be active at any given time.
+contract WorkflowRegistryManager is Ownable2StepMsgSender, ITypeAndVersion {
+  string public constant override typeAndVersion = "WorkflowRegistryManager 1.0.0";
+  uint8 private constant MAX_PAGINATION_LIMIT = 100;
+
+  struct Version {
+    address contractAddress; // ─╮ Address of the WorkflowRegistry contract
+    uint64 chainID; //           │ Chain ID of the EVM chain where the WorkflowRegistry is deployed.
+    uint32 deployedAt; // ───────╯ Block timestamp of deployment (sufficient until year 2106).
+    string contractTypeAndVersion; // WorkflowRegistry's typeAndVersion.
+  }
+
+  /// @notice Maps version numbers to their corresponding `Version` details.
+  /// @dev This mapping is 1-based, meaning version numbers start from 1. Ensure that all operations account for this
+  /// indexing strategy to avoid off-by-one errors.
+  mapping(uint32 versionNumber => Version versionInfo) private s_versions;
+
+  /// @notice The version number of the currently active WorkflowRegistry.
+  /// @dev Initialized to `type(uint32).max` to indicate no active version. Updated when a version is activated.
+  uint32 private s_activeVersionNumber = type(uint32).max;
```

Comment:
hm well, I thought the pattern of using type(uint32).max might be because changing something from 0 to something else cost a lot more than updating it from none zero. But I guess here specifically this would only apply once and there is also the deployment cost so it's not necessarily the best example of the right usecase.

> Initializing a storage variable is one of the most expensive operations a contract can do.When a storage variable goes from zero to non-zero, the user must pay 22,100 gas total (20,000 gas for a zero to non-zero write and 2,100 for a cold storage access).This is why the Openzeppelin reentrancy guard registers functions as active or not with 1 and 2 rather than 0 and 1. It only costs 5,000 gas to alter a storage variable from non-zero to non-zero.

Reply:
Readable and consistent code is more important than gas, especially gas costs that we pay once. And yes, you now pay it with the constructor instead of the first time use